### PR TITLE
Update dependencies + CI workflow, Fix metadata for Mod Menu

### LINKED ISSUE #23: Missing the blue client-only badge in modmenu
can you please modify the mod so that it gets properly recognized as client-only by modmenu?

### PR DESCRIPTION
**Pull request by Kichura**
1. This updates the `fabric.mod.json` file to fix a minor bug where the client badge is not correctly displayed inside the Mod Menu's listing system,
2. Updates dependencies in `gradle.properties` and the CI workflow to their latest versions,
3. Improves the CI workflow to be morely efficient,
4. Removes the unnecessary suggests context from the `fabric.mod.json`.

*(This superseds #24 and resolves #23, Please review this PR before merging)*

### CODE REVIEW

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "id": "forcecloseworldloadingscreen",
   "version": "${version}",
 
-  "name": "kennytv's epic force close loading screen mod for Fabric",
```
Comment (kennytv) [inline src/main/resources/fabric.mod.json:6]: no

Comment (Kichura) [reply src/main/resources/fabric.mod.json:6]: The problem with the previous name is that then the client tag cannot be displayed as intended.

Comment (kennytv) [reply src/main/resources/fabric.mod.json:6]: Well it's not a bug in *this* mod, and see https://github.com/kennytv/kennytvs-epic-force-close-loading-screen-mod-for-fabric/pull/24

Comment (Kichura) [reply src/main/resources/fabric.mod.json:6]: Will cancel my PR then.

**Pull request closed without merging**